cli/admin/src/main/java/io/pravega/cli/admin/AdminCommand.java

```diff
@@ -472,17 +478,21 @@ private String objectToJSON(Object object) {
         return new ObjectMapper().writeValueAsString(object);
     }
 
+    protected Controller instantiateController(ConnectionPool pool) {
+        return new ControllerImpl(ControllerImplConfig.builder()
+                                    .clientConfig(getClientConfig())
+                                    .build(), pool.getInternalExecutor());
+    }
+
     @VisibleForTesting
-    public SegmentHelper instantiateSegmentHelper(CuratorFramework zkClient) {
+    public SegmentHelper instantiateSegmentHelper(CuratorFramework zkClient, ConnectionPool pool) {
         HostControllerStore hostStore = createHostControllerStore(zkClient);
-        ConnectionPool pool = createConnectionPool();
         return new SegmentHelper(pool, hostStore, pool.getInternalExecutor());
     }
 
     @VisibleForTesting
-    public AdminSegmentHelper instantiateAdminSegmentHelper(CuratorFramework zkClient) {
+    public AdminSegmentHelper instantiateAdminSegmentHelper(CuratorFramework zkClient, ConnectionPool pool) {
         HostControllerStore hostStore = createHostControllerStore(zkClient);
-        ConnectionPool pool = createConnectionPool();
         return new AdminSegmentHelper(pool, hostStore, pool.getInternalExecutor());
```

Comment:
A close on the AdminSegmentHelper closes the connection pool.  Hence this change is not required.